Doc/library/test.rst

```diff
@@ -677,3 +677,182 @@ The :mod:`test.support` module defines the following classes:
+
+   There is also a special case for when *script_name* refers to a file
+   in a ``__pycache__`` directory. In that case, the helper will create a
+   new compiled bytecode file from the original source file using the
+   legacy naming scheme and place that in the zip archive instead.
+
+
+.. function:: make_pkg(pkg_dir, init_source='')
+
+   Creates a simple self-contained Python package, optionally with a
+   non-empty init file.
+
+   This function invalidates the import system caches, allowing the created
+   directory to be immediately imported.
+
+   Equivalent to::
+
+      os.mkdir(pkg_dir)
+      make_script(pkg_dir, "__init__", init_source)
+
+
+.. function:: make_zip_pkg(zip_dir, zip_basename, \
+                           pkg_name, script_basename, source, \
+                           depth=1, compiled=False)
+
+   Create a self-contained Python package inside a new zip archive in the
+   given directory.
+
+   This function invalidates the import system caches, allowing the contents
+   of the created archive to be immediately imported.
+
+   For example, this call::
+
+      make_zip_pkg('.', 'example', 'mypkg', 'submodule', '', depth=2)
+
+   would create an archive in the current directory called ``example.zip``
+   with the layout::
+
+      mypkg/
+        __init__.py
+        mypkg/
+          __init__.py
+          submodule.py
+
+   The init modules in the packages inside the archive are always empty,
+   while *source* is written to the submodule named by *script_basename*.
+
+   The *depth* argument controls how deeply nested the package is.
+
+   If *compiled* is true, then the files added to the archive will be
+   suitably-named compiled bytecode files rather than the original source
+   files.
+
+
+.. function:: interpreter_requires_environment()
+
+   Returns ``True`` if our :data:`sys.executable` interpreter requires
+   environment variables in order to be able to run at all.
+
+   This is designed to be used with :func:`unittest.skipIf` to annotate tests
+   that need to use an ``assert_python_*()`` function to launch an isolated
+   mode (-I) or no environment mode (-E) sub-interpreter process.
+
+   A normal build & test does not run into this situation but it can happen
+   when trying to run the standard library test suite from an interpreter that
+   doesn't have an obvious home with Python's current home finding logic.
+
+   Setting ``PYTHONHOME`` is one way to get most of the test suite to run in that
+   situation.  ``PYTHONPATH`` or ``PYTHONUSERSITE`` are the other common
+   environment variables that might impact whether or not the interpreter
+   can start.
+
+
+.. function:: run_python_until_end(*args, **env_vars)
+
+   Starts a python interpreter process with *args* and environment variables
+   *env_vars*. Pipes are created for stdout, stdin and stderr.
+
+   Waits until the process completes and returns the runcode, stdout and
+   stderr as a `_PythonRunResult`.
```

Comment:
It returns a tuple containing the `_PythonRunResult` and the command executed (`cmd_line`).